vault/resource_consul_secret_backend_test.go

```diff
@@ -24,83 +29,128 @@ func TestConsulSecretBackend(t *testing.T) {
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "client_key", "UPDATED-FAKE-CLIENT-CERT-KEY-MATERIAL"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, path),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldDescription, "test description"),
+					resource.TestCheckResourceAttr(resourceName, "default_lease_ttl_seconds", "0"),
+					resource.TestCheckResourceAttr(resourceName, "max_lease_ttl_seconds", "0"),
+					resource.TestCheckResourceAttr(resourceName, "address", "consul.domain.tld:8501"),
+					resource.TestCheckResourceAttr(resourceName, "token", token),
+					resource.TestCheckResourceAttr(resourceName, "scheme", "https"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldLocal, "false"),
+					resource.TestCheckResourceAttr(resourceName, "ca_cert", "FAKE-CERT-MATERIAL"),
+					resource.TestCheckResourceAttr(resourceName, "client_cert", "UPDATED-FAKE-CLIENT-CERT-MATERIAL"),
+					resource.TestCheckResourceAttr(resourceName, "client_key", "UPDATED-FAKE-CLIENT-CERT-KEY-MATERIAL"),
+				),
+			},
+			testutil.GetImportTestStep(resourceName, false, "token", "ca_cert", "client_cert", "client_key"),
+		},
+	})
+}
+
+func TestConsulSecretBackend_Bootstrap(t *testing.T) {
+	backend := acctest.RandomWithPrefix("tf-test-backend")
+	resourceName := "vault_consul_secret_backend.test"
+	resourceRoleName := "vault_consul_secret_backend_role.test"
+
+	if !testutil.CheckTestVaultVersion(t, "1.11") {
+		t.Skipf("test requires Vault 1.11 or newer")
+	}
+
+	cleanup, consulConfig := consulhelper.PrepareTestContainer(t, "1.12.3", false, false)
+	t.Cleanup(cleanup)
+	consulAddr := consulConfig.Address()
+
+	resource.Test(t, resource.TestCase{
+		Providers:    testProviders,
+		PreCheck:     func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy: testAccConsulSecretBackendCheckDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testConsulSecretBackend_bootstrapConfig(backend, consulAddr),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, backend),
+					resource.TestCheckResourceAttr(resourceName, "address", consulAddr),
+				),
+			},
+			{
+				Config: testConsulSecretBackend_checkBootstrapConfig(backend, consulAddr),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceRoleName, consts.FieldName, "management"),
+					resource.TestCheckResourceAttr(resourceRoleName, consts.FieldBackend, backend),
+					resource.TestCheckResourceAttr(resourceRoleName, "consul_policies.#", "1"),
+					resource.TestCheckTypeSetElemAttr(resourceRoleName, "consul_policies.*", "global-management"),
 				),
 			},
+			testutil.GetImportTestStep(resourceName, false),
+			{
+				Config:      testConsulSecretBackend_bootstrapConfig(backend+"-new", consulAddr),
+				ExpectError: regexp.MustCompile(`Token not provided and failed to bootstrap ACLs`),
+			},
+			testutil.GetImportTestStep(resourceName, false),
```

Comment:
Suggest moving this to come after the first config test step.

Reply:
I actually have it here twice. This one and another a couple lines up after the checkBootstrapConfig, which is after the first config is made. Are you saying it should specifically be checked before the role is made in step 2?

Second question - when attempting to write the resource config the second time, it fails because Consul is already bootstrapped which is what the test wants. But the import afterwards is looking for that new resource path that didn't succeed. I would have guessed the error should stop Terraform from changing the state to the new path. Is this expected behavior though?

Reply:
> I actually have it here twice. This one and another a couple lines up after the checkBootstrapConfig, which is after the first config is made. Are you saying it should specifically be checked before the role is made in step 2?

Essentially yes, see suggestion above.

> 
> Second question - when attempting to write the resource config the second time, it fails because Consul is already bootstrapped which is what the test wants. But the import afterwards is looking for that new resource path that didn't succeed. I would have guessed the error should stop Terraform from changing the state to the new path. Is this expected behavior though?

Yes. since the "new" path was never  created the import test step should fail since the Vault is not provisioned with that path.